src/extension/background-script/state.js

```diff
@@ -0,0 +1,62 @@
+import browser from "webextension-polyfill";
+import createState from "zustand";
+
+// these keys get synced from the state to the browser storage
+// the values are the default values
+const browserStorage = {
+  settings: {},
+  accounts: {},
+  allowances: [],
+  currentAccountId: null,
+};
+
+const state = createState((set, get) => ({
+  connector: null,
+  account: null,
+  settings: {},
+  accounts: {},
+  allowances: [],
+  currentAccountId: null,
+  password: null,
+  lock: () => set({ password: null, connector: null, account: null }),
+  init: () => {
+    return browser.storage.sync
```

Comment:
Sync storage can be updated from other instances of a browser so you might want to use `browser.storage.onChanged` to avoid getting the state out of sync. 

A bit rare, but this might happen if, say, I leave the browser always open at home, change something to my extension at work, and come back to it at night.

Reply:
very true. should add an onchanged handler to update the state in memory. 👍